app/src/androidTest/java/org/mozilla/fenix/ui/robots/HistoryRobot.kt

```diff
@@ -1,20 +1,92 @@
-import androidx.test.espresso.matcher.ViewMatchers.withId
-import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
-import androidx.test.espresso.matcher.ViewMatchers.Visibility
+import androidx.test.espresso.action.ViewActions.longClick
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.RootMatchers.isDialog
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.matcher.ViewMatchers.isDescendantOfA
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
 import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withParent
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.Until
+import org.hamcrest.Matchers
 import org.hamcrest.Matchers.allOf
 import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.TestAssetHelper.waitingTime
 import org.mozilla.fenix.helpers.click
 
+/**
+ * Implementation of Robot Pattern for the history menu.
+ */
 class HistoryRobot {
 
-    fun verifyHistoryMenuView() = assertHistoryView()
+    fun verifyHistoryMenuView() = assertHistoryMenuView()
+
+    fun verifyEmptyHistoryView() = assertEmptyHistoryView()
+
+    fun verifyVisitedTimeTitle() = assertVisitedTimeTitle()
+
+    fun verifyFirstTestPageTitle(title: String) = assertTestPageTitle(title)
+
+    fun verifyTestPageUrl(expectedUrl: Uri) = assertPageUrl(expectedUrl)
+
+    fun verifyDeleteConfirmationMessage() = assertDeleteConfirmationMessage()
+
+    fun verifyMultiSelectionCheckmark() = assertMultiSelectionCheckmark()
+
+    fun verifyMultiSelectionCounter() = assertMultiSelectionCounter()
+
+    fun verifyShareButton() = assertShareButton()
+
+    fun verifyShareOverlay() = assertShareOverlay()
+
+    fun verifyShareTabFavicon() = assertShareTabFavicon()
+
+    fun verifyShareTabTitle() = assertShareTabTitle()
+
+    fun verifyShareTabUrl() = assertShareTabUrl()
+
+    fun openOverflowMenu() {
+        overflowMenu().click()
+    }
+
+    fun clickThreeDotMenuDelete() {
+        threeDotMenuDeleteButton().click()
+    }
+
+    fun clickDeleteHistoryButton() {
+        deleteAllHistoryButton().click()
+    }
+
+    fun confirmDeleteAllHistory() {
+        onView(withText("Delete"))
+            .inRoot(isDialog())
+            .check(matches(isDisplayed()))
+            .click()
+    }
+
+    fun clickShareButton() {
+        shareButton().click()
+        mDevice.wait(Until.findObject(By.text("SHARE A LINK")), waitingTime)
+    }
+
+    fun longTapSelectItem(url: Uri) {
+        onView(withText(url.toString())).perform(longClick())
+    }
+
+    fun clickMultiSelectionDelete() {
+        multiSelectionDeleteButton().click()
+    }
```

Comment:
perhaps with this as well, if it can be more generic

Reply:
this one I don't know if it can be a generic method, but I moved everything related to the multiple selection toolbar to a new Robot and they can be reused on Bookmarks (and maybe other lists in the future, like downloads or logins).